core/modules/wsched/BlendScheduler.cc

```diff
@@ -137,6 +137,134 @@ void BlendScheduler::_sortScanSchedulers() {
+        }
+
+        util::LockGuardTimed guard(util::CommandQueue::_mx, "BlendScheduler::queCmd b");
+        LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd e");
+        // Check for scan tables. The information for all tasks should be the same
+        // as they all belong to the same query, so only examine the first task.
+        if (first) {
+            LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd f");
+            first = false;
+
+            auto const& scanTables = task->getScanInfo().infoTables;
+            bool interactive = task->getScanInteractive();
+            if (scanTables.size() <= 0 || interactive) {
+                LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd g");
+                // If there are no scan tables, no point in putting on a shared scan.
+                LOGS(_log, LOG_LVL_DEBUG, "Blend chose group scanTables.size=" << scanTables.size()
+                        << " interactive=" << interactive);
+                //&&&task->setOnInteractive(true);
+                onInteractive = true;
+                s = _group;
+            } else {
+                LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd h");
+                //&&&task->setOnInteractive(false);
+                onInteractive = false;
+                int scanPriority = task->getScanInfo().scanRating;
+                if (LOG_CHECK_LVL(_log, LOG_LVL_DEBUG)) {
+                    ostringstream ss;
+                    ss << "Blend chose scan for priority=" << scanPriority << " : ";
+                    for (auto scanTbl : scanTables) {
+                        ss << scanTbl.db + "." + scanTbl.table + " ";
+                    }
+                    LOGS(_log, LOG_LVL_DEBUG, ss.str());
+                }
+                LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd i");
+                {   // Find the scheduler responsible for this 'scanPriority'.
+                    lock_guard<mutex> lg(_schedMtx);
+                    LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd j");
+                    for (auto const& sched : _schedulers) {
+                        LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd j1");
+                        ScanScheduler::Ptr scan = dynamic_pointer_cast<ScanScheduler>(sched);
+                        if (scan != nullptr) {
+                            if (scan->isRatingInRange(scanPriority)) {
+                                s = scan;
+                                break;
+                            }
+                        }
+                    }
+                }
+                LOGS(_log, LOG_LVL_WARN, "&&& BlendScheduler::queCmd k");
+                // If the user query for this task has been booted, put this task on the snail scheduler.
+                auto queryStats = _queries->getStats(task->getQueryId());
+                if (queryStats && queryStats->getQueryBooted()) {
+                    s = _scanSnail;
```

Comment:
It looks like a value of `s` that might be (if certain conditions were met) previously set in the `for` loop above would be overwritten here. Would this cause any issues?

Reply:
It is supposed to be overwritten. The point of the snail scheduler is that user queries that take too long get moved to the snail  scheduler, which is low priority.